src/main/java/Main.java

```diff
@@ -49,27 +46,20 @@ private static void waitForTargetHealthcheck() throws InterruptedException {
     }
 
     private static Disposable createConsumer(MonitoringServer monitoringServer) {
-        return new Consumer(
-            new ReactiveKafkaClient<>(
-                KafkaClientFactory.createConsumer(),
-                topicsRoutes.getTopics(),
-                new ConsumerRebalanceListener() {
-                    @Override
-                    public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
-                        if (partitions.size() == 0) {
-                            return;
-                        }
-
-                        Monitor.assignedToPartition(partitions);
-                        monitoringServer.consumerAssigned();
-                    }
-
-                    @Override
-                    public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
-                        Monitor.revokedFromPartition(partitions);
-                    }
+        var receiverOptions = KafkaClientFactory
+            .createReceiverOptions()
+            .subscription(topicsRoutes.getTopics())
+            .addAssignListener(partitions -> {
+                if (partitions.isEmpty()) {
```

Comment:
if partitions are empty it might indicate something wrong, probably better to log it

Reply:
I don't want us to miss something because of this

Reply:
I moved the log `Monitor.assignedToPartition(partitions)` to be the first line in this handler.
It's interesting to see if we see 0 partitions assigned.
Honestly I have no idea why we put this here long time ago.. If the logs will show that this use case doesn't happen I'll remove the check altogether